### PR TITLE
Fix settings

### PR DESCRIPTION
**Pull request by raphael-pietrzak**
## Summary

This PR fixes several issues and enhances the application as follows:

- Implements a new SFX slider in settings.
- Fixes background transparency issues.
- Updates the selected color.
- Resets height and selection in descriptions.

## Checklist

- [x] I have tested this change locally and it works as expected.
- [x] I have made sure that the code follows the formatting and style guidelines of the project.

## Labels
`type: fix`, `area: settings`, `area: color`, `area: description`

### CODE REVIEW

**Comment by raphael-pietrzak** [comment]
I have read Contributing file about ruff but the command line don't even works on main branch.

Also I  use surface.fill('green') in my code because pygame.SRCALPHA is not enough because i blit a moving surface on a surface.

**Comment by fortwoone** [comment]
> I have read Contributing file about ruff but the command line don't even works on main branch.

Read `README.md`. You forgot to install the dev requirements.

> 
> Also I  use surface.fill('green') in my code because pygame.SRCALPHA is not enough because i blit a moving surface on a surface.

...what the hey do you mean? Instead of filling it with green and wasting a colour, using `pygame.SRCALPHA` lets you just fill the surface with the transparent colour instead

File: src/gui/menu/description.py [hunk 1]
```diff
@@ -30,13 +30,23 @@ def setup(self):
         self.description_surface = pygame.Surface(
             self.description_rect.size, pygame.SRCALPHA
         )
+        self.description_surface.fill('green')
+        self.description_surface.set_colorkey('green')
```
Comment (fortwoone) [inline src/gui/menu/description.py:34]: Not necessary

File: src/gui/menu/description.py [hunk 2]
```diff
@@ -57,7 +67,8 @@ def mouse_wheel(self, event) -> bool:
 
     # draw
     def make_surface_transparent(self):
-        self.description_surface.fill(pygame.Color(0, 0, 0, 0))
+        self.description_surface.fill('green')
+        self.description_slider_surface.fill('green')
```
Comment (fortwoone) [inline src/gui/menu/description.py:71]: Same comment

File: src/gui/menu/description.py [hunk 3]
```diff
@@ -74,8 +85,16 @@ def draw_slider_bar(self):
         pygame.draw.rect(self.display_surface, "grey", slide_bar_rect, 0, 4)
 
     def draw(self):
-        pygame.draw.rect(self.display_surface, "White", self.rect, 0, 4)
-        self.make_surface_transparent()
```
Comment (fortwoone) [inline src/gui/menu/description.py:78]: Of course you'll get rendering artifacts if you remove this call...

**Review by fortwoone — CHANGES_REQUESTED** [review]

**Comment by raphael-pietrzak** [comment]
> Read `README.md`. You forgot to install the dev requirements.

I installed it, maybe i can give you the full traceback on discord ?

> ...what the hey do you mean? Instead of filling it with green and wasting a colour, using `pygame.SRCALPHA` lets you just fill the surface with the transparent colour instead

Using `pygame.SRCALPHA` don't erase buttons animations because it's just transparent surface instead of coloring over 'green' and delete 'green'. By the way, this was the settings display before my changes :


Screenshot 2024-08-03 at 13 32 02

File: src/gui/menu/description.py (hunk 3 above)
Comment (raphael-pietrzak) [reply src/gui/menu/description.py:78]: I moved the call on children classes

File: src/gui/menu/description.py (hunk 1 above)
Comment (raphael-pietrzak) [reply src/gui/menu/description.py:34]: I get it but why pygame.SRCALPHA doesn't work so ?

Comment (fortwoone) [reply src/gui/menu/description.py:34]: The returned surface is already transparent when created

File: src/gui/menu/description.py (hunk 2 above)
Comment (fortwoone) [reply src/gui/menu/description.py:71]: As for all further instances of this, you need to fill the images with the transparent colour first. If we were to decide using green for the sliders, we would get an unexpected rendering fault due to the colourkey existing, which is why I prefer using alpha channels instead

```suggestion
        self.description_surface.fill(pygame.Color(0, 0, 0, 0))
        self.description_slider_surface.fill(pygame.Color(0, 0, 0, 0))
```

**Comment by fortwoone** [comment]
> > Read README.md. You forgot to install the dev requirements.
> 
> I didn't touch anything but today it's working 😂
> 
> > ...what the hey do you mean? Instead of filling it with green and wasting a colour, using `pygame.SRCALPHA` lets you just fill the surface with the transparent colour instead
> 
> Using `pygame.SRCALPHA` don't erase buttons animations because it's just transparent surface instead of coloring over 'green' and delete 'green'. By the way, this was the settings display before my changes :
> 
> Screenshot 2024-08-03 at 13 32 02

I found out why it didn't work, actually, and it's because of you using the colourkey. The culprit is `KeybindsDescription.create_keybinds`.

```py
    def create_keybinds(self):
        margin = 10
        size = (600, 60 * self.controls.length() + 2 * margin)
        self.description_slider_surface = pygame.Surface((size))  # because here you still use the colourkey while the rest of the code expects alpha support to be enabled on this surface
        self.description_slider_rect = self.description_slider_surface.get_rect()
        self.description_slider_surface.set_colorkey("green")

        self.keys_group.clear()
        index = 0
        for control in self.controls.all_controls():
            name = self.controls(control).name

            unicode = self.value_to_unicode(control.control_value)
            path = self.get_path(control.control_value)

            image = pygame.image.load(path)
            image = pygame.transform.scale(image, (40, 40))

            topleft = (10, 10 + 60 * index)
            key_setup_button = KeySetup(name, control, unicode, topleft, image)
            self.keys_group.append(key_setup_button)

            index += 1
```

File: pyproject.toml [hunk 4]
```diff
@@ -31,7 +31,7 @@ description = "A game based on Pydeew Valley by ClearCode, used by the Universit
 # Use Ruff to replace Black, isort, and Flake8
 [tool.ruff]
 line-length = 88
-target-version = "py312"
+target-version = "py38"
```
Comment (Mega-JC) [inline pyproject.toml:34]:
```suggestion
target-version = "py312"
```

Why did you modify this? This project is using Python 3.12. You fix linting and formatting errors by running the scripts and making changes, not changing the settings.

**Review by Mega-JC — CHANGES_REQUESTED** [review]

**Comment by raphael-pietrzak** [comment]
> I found out why it didn't work, actually, and it's because of you using the colourkey. The culprit is `KeybindsDescription.create_keybinds`.

you're right, this resolve the black background. I am still wondering how erase button animation after selection using an other way than using `fill('green')` :

Screenshot 2024-08-04 at 02 02 19

I will take back as it was before with only `SRCALPHA`. 
This isn't an high priority task anyway I better concentrate myself on useful tasks

**Review by fortwoone — APPROVED** [review]

**Pull request merged**